src/marten/db/query/sql/query.cr

```diff
@@ -74,6 +74,25 @@ module Marten
             ensure_join_for_field_path(field_path, selected: true)
           end
 
+          def average(raw_field : String)
+            field_context = get_field_context(raw_field, Model)
+
+            unless field_context.field.is_a?(Marten::DB::Field::Int) ||
+                   field_context.field.is_a?(Marten::DB::Field::Float)
+              raise Errors::InvalidField.new(
+                "Cant calculate the average of '#{raw_field}' (#{field_context.field.class})."
+              )
+            end
+
+            column_name = solve_field_and_column(raw_field).last
+
+            sql, parameters = build_average_query(column_name)
+            connection.open do |db|
+              result = db.scalar(sql, args: parameters)
+              result ? result.to_s.to_f : 0.0
```

Comment:
Shouldn't we return `nil` in case no records are matched by the query set? That way it would be easier to differentiate cases where indeed the average is `0.0` to cases where no records are matched by the query set.